### PR TITLE
III-5887 - Add DeleteCurrentOrganizerHandler to catch Full updates of Events with TicketSales

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

- `CommandHandler\DeleteCurrentOrganizerHandler` & test: Added to inject `EventHasTicketSalesGuard`

### Changed

- `OfferServiceProvider.php`: add `DeleteCurrentOrganizerHandler` to `container`

- `CommandBusServiceProvider`: subscribe `CommandBus` to `DeleteCurrentOrganizerHandler`

- `OfferCommandHelper` & Children `EventCommandHelper`, `PlaceCommandHelper` & `ItemCommandHelper`: remove `getDeleteCurrentOrganizerClassName()` `handleDeleteCurrentOrganizer()`

- Made `Offer\DeleteCurrentOrganizer` `final` instead of `abstract`

- refacored `tests`

### Removed

- No longer needed `childClasses` of `Offer\DeleteCurrentOrganizer`

### Fixed

- One can no longer delete the `Organizer`, when an `Event` has `TicketSales` during a full-update

- Some boyScouting, made `Classes` final where appropriate.

---
Ticket: https://jira.publiq.be/browse/III-5887

### CODE REVIEW

File: src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        OfferRepository $offerRepository,
+        EventHasTicketSalesGuard $organizerChangeAllowedBasedOnTicketSales
+    ) {
+        $this->offerRepository = $offerRepository;
+        $this->organizerChangeAllowedBasedOnTicketSales = $organizerChangeAllowedBasedOnTicketSales;
+    }
+
+    public function handle($command): void
+    {
+        if (!($command instanceof DeleteCurrentOrganizer)) {
```
Comment (grubolsch) [inline src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php:28]: Technicly you don't need the inner - (), instanceof has a higher priority than the !.

File: src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            return;
+        }
+
+        $offer = $this->offerRepository->load($command->getItemId());
```
Comment (grubolsch) [inline src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php:32]: Are we sure this always gives an offer back? load() can throw an EventStreamNotFoundException, should we handle that here?

**Review by grubolsch — APPROVED** [review]

File: src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php (hunk 1 above)
Comment (JonasVHG) [reply src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php]: Nicely spotted.

File: src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php (hunk 2 above)
Comment (JonasVHG) [reply src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php:32]: Yes, if you include an unknown `id`, a new `Offer` is created, and you cannot delete an Organizer during a `create` only during an `update` so the `OfferRepository` should always return an `Offer`.
Whether this logic desirable is another matter, but I thought that was for legacy reasons like `OMD`.

Comment (LucWollants) [reply src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php:32]: Throwing the exception `AggregateNotFoundException` is already handled inside the `ApiProblemFactory`. Inside the command handlers, there is no guarantee that the ID passed is that of an existing item.

**Pull request merged**